enhancements/machine-config/mco-network-configuration.md

```diff
@@ -0,0 +1,253 @@
+
+### Advantages
+* Rollout keeps the same semantics as MCO. No longer a network state can end up with all the nodes losing connectivity at once.
+* Kubernetes-NMState Rollback functionality is kept (When it is triggered, it ends up in [errUnreconcilable](https://github.com/openshift/machine-config-operator/blob/15edac1b5613b135973b071cf96535c3a376419f/pkg/daemon/update.go#L193)).
+* MCO keeps ownership over node configuration.
+* API consistency and granularity for node changes is kept.
+* Customers can keep using declarative Network configuration.
+
+### Disadvantages
+* Deviates from upstream kubevirt knmstate usage. [midstream](https://github.com/openshift/kubernetes-nmstate) needs to develop:
+    * A very thin controller that creates MC objects
+    * Expose an interface for MCD to keep **NNS** up to date with the OpenShift API.
+    * Since /usr/libexec/platform-python is on the way out, we must run **NMState with podman**.
+* MCD needs to make a change in its [update](https://github.com/openshift/machine-config-operator/blob/15edac1b5613b135973b071cf96535c3a376419f/pkg/daemon/update.go#L152) logic to detect if only networkState changed and in such case, use the vendored Kubernetes NMState functionality instead of *updateOSAndReboot*.
+* While the API can be deemed Tech Preview, it would effectively be baked-in, so it can't be made optional to deploy without it.
+* NodeNetworkState and NodeNetworkConfigPolicy probably need to go into [openshift/api](https://github.com/openshift/api).
+* Users lose the upstream ability to target specific single nodes in favor of MachinePool granularity.  (walters: See also https://github.com/openshift/machine-config-operator/issues/1720)
+
+### Open Questions [optional]
+
+### Test Plan
+
+- Automated CI tests
+
+### Graduation Criteria
+
+- TBD
+
+#### Examples
+
+
+##### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+##### Tech Preview -> GA 
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+##### Removing a deprecated feature
+
+- Announce deprecation and support policy of the existing feature
+- Deprecate the feature
+
+### Upgrade / Downgrade Strategy
+
+An upgrade path from a user consuming the upstream kubernetes-nmstate component should be researched and implemented.
+
+### Version Skew Strategy
+
+The MCO integration should be able to operate independently of the kubernetes-nmstate version. The kubernetes-nmstate component itself will have a dependency on the nmstate version itself, which has a tie to NetworkManager version.
+
+## Implementation History
+
+## Drawbacks
+
+## Alternatives
+
+### Option A
+
+Include kubernetes-nmstate as a standalone component within OpenShift
```

Comment:
Can we hear a little about why this would be a worse choice?

Reply:
We had a previous OEP[1] that was basically rejected for the reason of "we already have a component that does this" and were told that component was MCO. As you can see, that isn't exactly the case.

[1] #161